api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoTrimestralCohortQueries.java

```diff
@@ -83,8 +90,23 @@ public CohortDefinition getE() {
    *     treatment who received one Viral load result
    */
   public CohortDefinition getF() {
-    AllPatientsCohortDefinition cd = new AllPatientsCohortDefinition();
-    cd.setParameters(getParameters());
+    CohortDefinition cohortE = getE();
+    SqlCohortDefinition sqlCohortDefinition = new SqlCohortDefinition();
+    sqlCohortDefinition.setName(
+        "Patients in the 1st line treatment who received one Viral load result");
+    sqlCohortDefinition.addParameter(new Parameter("onOrAfter", "onOrAfter", Date.class));
+    sqlCohortDefinition.addParameter(new Parameter("onOrBefore", "onOrBefore", Date.class));
+    sqlCohortDefinition.addParameter(new Parameter("location", "location", Location.class));
+    sqlCohortDefinition.setQuery(
+        ResumoTrimestralQueries.getPatientsWhoReceivedOneViralLoadResult(
+            hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId(),
+            hivMetadata.getHivViralLoadConcept().getConceptId(),
+            hivMetadata.getBeyondDetectableLimitConcept().getConceptId()));
+
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+    cd.addSearch("E", mapStraightThrough(cohortE));
+    cd.addSearch("FI", mapStraightThrough(sqlCohortDefinition));
+    cd.setCompositionString("E AND NOT FI");
```

Comment:
Please call this E _cohortE_ and sqlCohortDefinition call it _viralLoadResult_

Reply:
Done